esp8266-fastled-webserver/common.h

```diff
@@ -0,0 +1,336 @@
+/*
+   ESP8266 FastLED WebServer: https://github.com/jasoncoon/esp8266-fastled-webserver
+   Copyright (C) Jason Coon
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#if !defined(ESP8266_FASTLED_WEBSERVER_COMMON_H)
+#define ESP8266_FASTLED_WEBSERVER_COMMON_H
+
+#include "Arduino.h"
+
+// config.h must be included before FastLED, as it may define FastLED options
+// such as whether to disable interrupts, etc.
+// config.h defines pre-processor symbols, but does not itself define functions, structs, etc.
+#include "config.h"
+
+#if 1 // external libraries
+  #define FASTLED_INTERNAL // no other way to suppress build warnings
+  #include <FastLED.h>
+  FASTLED_USING_NAMESPACE
+
+  extern "C" {
+  #include "user_interface.h"
+  }
+
+  #include <LittleFS.h>
+  #define MYFS LittleFS
+
+  #include <NTPClient.h>
+  #include <ESP8266WiFi.h>
+  #include <WiFiUdp.h>
+  #include <ESP8266mDNS.h>
+  #include <ESP8266WebServer.h>
+  #include <ESP8266HTTPUpdateServer.h>
+  #include <ESP8266HTTPClient.h>
+  //#include <WebSocketsServer.h>
+  #include <EEPROM.h>
+  #include <WiFiManager.h> // https://github.com/tzapu/WiFiManager/tree/development
+
+
+  #include "./include/simplehacks/static_eval.h"
+  #include "./include/simplehacks/constexpr_strlen.h"
+  #include "./include/simplehacks/array_size2.h"
+#endif // 1
+
+void dimAll(byte value);
+// info.cpp
+String WiFi_SSID(bool persistent);
+String getInfoJson();
+
+
+// Ugly macro-like constexpr, used for FastLED template arguments
+template <size_t ONE_BASED_OUTPUT_CHANNEL>
```

Comment:
This template function may look ugly, but it allows `constexpr` calculation of the start and end of any strand of pixel, when using a multiple-channel output (e.g., ESPTHING and Fibonacci512).  Thus, ZERO computation is required at runtime for these values.